### ISSUE TITLE
[java] PrematureDeclaration false positive for unused variables

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3221]
**Affects PMD Version:** 6.33.0

**Rule:** [PrematureDeclaration](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#prematuredeclaration)

**Description:**

Don't report variables that are not used at all. Those should be left to [UnusedLocalVariable](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable).

**Code Sample demonstrating the issue:**

```java
public String wrongEscapeSequences() 
    {
        final String r1 = "\u0008"; // unused var - not a premature declaration
        return "\u000csssdfsd";
    }
```

(from https://github.com/checkstyle/checkstyle/blob/checkstyle-8.10/src/it/resources/com/google/checkstyle/test/chapter2filebasic/rule232specialescape/InputIllegalTokenText.java#L17)

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
